### PR TITLE
Config to skip OpenTelemetry.propagation.inject(metadata)

### PR DESCRIPTION
**Pull request by tgwizard**
An alternative to https://github.com/hibachrach/opentelemetry-instrumentation-grpc/pull/3 that's probably cleaner.

### CODE REVIEW

File: lib/opentelemetry/instrumentation/grpc/interceptors/client.rb [hunk 1]
```diff
@@ -48,7 +48,7 @@ def call(type:, requests: nil, call: nil, method: nil, metadata: nil)
               attributes: attributes,
               kind: OpenTelemetry::Trace::SpanKind::CLIENT
             ) do |span|
-              OpenTelemetry.propagation.inject(metadata)
+              OpenTelemetry.propagation.inject(metadata) unless instrumentation_config[:skip_telemetry_propagation]
```
Comment (fbogsany) [inline lib/opentelemetry/instrumentation/grpc/interceptors/client.rb:51]: 👍

Comment (dtwardowski) [reply lib/opentelemetry/instrumentation/grpc/interceptors/client.rb:51]: @hibachrach @tgwizard Will this change be part of any upcoming release? If so, when can we expect it?

Comment (hibachrach) [reply lib/opentelemetry/instrumentation/grpc/interceptors/client.rb:51]: So sorry, I have no idea why I didn't see this PR, but I plan on archiving this repo soon as this code has merged into [`open-telemetry/opentelemetry-ruby-contrib`](https://github.com/open-telemetry/opentelemetry-ruby-contrib). Please feel free to open a PR there (in accordance with their contribution guidelines, of course).

Comment (tgwizard) [reply lib/opentelemetry/instrumentation/grpc/interceptors/client.rb:51]: Will do, thanks.

**Pull request closed without merging**